js/oyst/oneclick.js

```diff
@@ -11,10 +11,18 @@
 /**
  * OneClick Customs JS
  */
+"use strict";
 
 function showMessage(txt, type) {
-    var html = '<ul class="messages"><li class="' + type + '-msg"><ul><li>' + txt + '</li></ul></li></ul>';
+    var html = "<ul class=\"messages\"><li class=\"" + type + "\"-msg\"><ul><li>" + txt + "</li></ul></li></ul>";
     $("messages").update(html);
-    var url = location.href.split('#')[0]
-    location.replace(url + "#html-body");
+    var url = window.location.href.split("#")[0];
+    window.location.replace(url + "#html-body");
+}
+
+if (Validation) {
+    Validation.add('validate-oyst-url', 'Please enter a valid URL. Protocol is required (http:// or https://), finishing by "/"', function (v) {
+        v = (v || '').replace(/^\s+/, '').replace(/\s+$/, '');
```

Comment:
isn't it `trim` function job?

Reply:
`validate-oyst-url ` validator is an adaptation of https://github.com/OpenMage/magento-mirror/blob/d48bebc211cc216aaf78bdf25d7f0b0143d6333b/js/prototype/validation.js#L576